Add strapping pin information to FAQ (#1576)
OttoWinter/esphomedocs: guides/faq.rst

Important: follow these `instructions </components/esphome.html#changing-esphome-node-name>`_ to use the ``use_address`` parameter when renaming a live device, as the connection to an existing device will only work with the old name until the name change is complete.

Why am I getting a warning about strapping pins?
--------------------------------------------------

> The strapping pins are read on esp bootup and determine how it boots up. The define flashing mode, normal boot and a couple of other settings. 

Reviewer: ```suggestion
The ESP chips have special "strapping pins" that are read during the bootup procedure and determine how it boots up. They define whether the ESP boots into a special "flashing mode" or normal boot and a couple of other internal settings.
```

> If an external pullup/down changes the configured voltage levels boot failure or hard to diagnose issues can happen.  

Reviewer: ```suggestion
If an external pullup/down changes the configured voltage levels boot failures or hard to diagnose issues can happen.
```

> While the use of them in software is not a problem if  there's something attached to the pins (particularly if they're not floating) you may run into problems.

Reviewer: ```suggestion
While the use of them in software is not a problem, if there's something attached to the pins (particularly if they're not floating during the bootup) you may run into problems.
```

> Unless you have a pressing need to use them and you know exactly what you are doing it is recommended to avoid them. 

Reviewer: ```suggestion
It's recommended to avoid them unless you have a pressing need to use them and you know have reviewed the expected boot voltage levels of these pins from the ESP datasheet.
```

> 

Reviewer: ```suggestion

Note that some boards connect pins such as GPIO0 to a builtin tactile switch. In these cases using the strapping pins is not a problem.
```

(on a couple of ESP32s I have, GPIO0 is connected to the flash pin which you can also use as a button yourself; For those cases I think it makes sense to explicitly state that their use is ok)
